qemu: Add LOAD_IMAGE_V2 support

tekkamanninja opened the pull request:
The generic LOAD_IMAGE_V2 framework has been merged and enable for almost
all the arm platform. Because qemu platform doesn't share those common
files with arm, QEMU haven't got this support yet.

This patch add all the necessary code the files for adding LOAD_IMAGE_V2
support on QEMU and enable it as default.

Fixes ARM-software/tf-issues#507

Signed-off-by: Fu Wei <fu.wei@linaro.org>

davidcunado-arm commented:
@tekkamanninja 
FYI - as per the contributing guidelines, PRs should be made against integration. I've already changed this PR, so action needed in that respect.

davidcunado-arm commented:
jenkins: test this please

tekkamanninja commented:
@davidcunado-arm 
Great thanks for your help, :-)

tekkamanninja commented:
Can I have the details of the failed check, then I can try to fix it.
@davidcunado-arm

davidcunado-arm commented:
@tekkamanninja 
The check fails were to do with the copyrights.

tekkamanninja commented on plat/qemu/qemu_image_load.c:
```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2017, Linaro Limited. All rights reserved.
```
@davidcunado-arm 
Since I am a Linaro assignee, So I just check the existing file for the Copyright, I found a services/spd/opteed/teesmc_opteed.h  has this. So I copy and paste it.
So I should use 
Copyright (c) 2017, ARM Limited and Contributors. All rights reserved.
Right??

danh-arm replied:
Correct. The "ARM Limited and Contributors" boilerplate code should be used for all contributions except imported 3rd party code (which this isn't). That teesmc_opteed.h header is a bit special as it contains code from a Linaro maintained project (OPTEE), which was modified and contributed here.

tekkamanninja replied:
OK, fixing it now

tekkamanninja commented:
@davidcunado-arm 
I have updated the commit, please review it, great thanks for your correction

davidcunado-arm commented:
jenkins: test this please

davidcunado-arm commented:
@jenswi-linaro 
Could you ACK this change for qemu?

jenswi-linaro commented:
This fails to boot in my test setup. Before this patch it boots OK so clearly this patch breaks something.

tekkamanninja commented:
@jenswi-linaro 
I just tried it, in my test setup, it works(boot into Linaro rootfs).
Would you mind to let me know your test setup to help me to debug it..
I guess the problem is about BL2 or FIP.

My test setup is here:
cd /home/tekkamanninja/development/Tekkaman_SDK/output/arm-trusted-firmware_tekkaman
/home/tekkamanninja/development/Tekkaman_SDK/output/qemu_ras/bin/qemu-system-aarch64  -cpu cortex-a57  -smp 2  -m 2048  -machine virt,secure=on,virtualization=on   -nographic  -bios /home/tekkamanninja/development/Tekkaman_SDK/output/arm-trusted-firmware_tekkaman/bl1.bin  -drive if=none,file=/home/tekkamanninja/development/Tekkaman_SDK/validation/LEG/vexpress64-leg-sd.img,id=vda,cache=writeback -device virtio-blk-device,drive=vda    -serial telnet:localhost:7000,server  -d unimp -semihosting-config enable,target=native
QEMU 2.9.90 monitor - type 'help' for more information
(qemu)

tekkamanninja commented:
@jenswi-linaro Great thanks for your info, I have updated the commit, please help me to review it.
I have reproduced the problem(BL32 loading issue), and fixed it. Great thanks for pointing out the bug.

jenswi-linaro commented:
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

danh-arm commented:
jenkins:retest this please
